wip-0018.md

```diff
@@ -25,15 +25,69 @@ format!("inner: {:?}", inner)
 
 In other words, it creates a string that contains the debug representation of the inner error. The debug representation is created by the Rust compiler, and it may change between Rust versions. And the inner error is supposed to be an implementation detail of witnet-rust, so it doesn't make much sense for it to be part of consensus-critical code. This gives us a good reason to remove this message.
 
+This message may also appear during the retrieval and aggregation stages. In that case the revealed bytes can be different for different witnesses, so there it is not consensus-critial and changing the message is not a breaking change. Therefore the message argument should only be removed in the tally stage, when serializing the result of the tally script.
+
+### Example
+
+There are two kinds of `UnhandledIntercept` errors, depending on where they are created:
+
+* Created in tally
+
+* Created in retrieval/aggregation
+
+#### Created in tally
+
+This document focuses on those errors that are created in the tally stage, because they are the consensus-critial ones. As an example, see data request `63c1ff0bacd24676bd71fa2bb984e8faee22cd8c73b14b83c57149a792dc315c`. In that request every node reports a different string, and the mode operator returns an error because there is no "most common string". This is the tally using the old validation rules:
```

Comment:
```suggestion
This document focuses on those errors that are created in the tally stage because they are the consensus-critical. As an example, see data request `63c1ff0bacd24676bd71fa2bb984e8faee22cd8c73b14b83c57149a792dc315c`. In that request, every node reports a different string, and the mode operator returns an error because there is no "most common string". This is the tally using the old validation rules:
```